src/datatype/ostree.rs

```diff
@@ -87,24 +96,47 @@ impl OstreePackage {
         }
     }
 
-    /// Get the current OSTree package based on the last successful installation
-    /// if it exists, or from running `ostree admin status` otherwise.
-    pub fn get_ecu(serial: &str) -> Result<OstreePackage, Error> {
-        let branch = if Path::new(NEW_PACKAGE).exists() {
-            debug!("getting ostree package from `{}`", NEW_PACKAGE);
-            return Ok(json::from_reader(BufReader::new(File::open(NEW_PACKAGE)?))?);
+    /// Install this package using the `ostree` command.
+    pub fn install(&self, creds: &Credentials) -> Result<InstallOutcome, Error> {
+        debug!("installing ostree commit {}", self.commit);
+        let from = Self::get_latest(&self.ecu_serial)?;
+        if from.commit == self.commit {
+            return Ok(InstallOutcome::new(InstallCode::ALREADY_PROCESSED, "".into(), "".into()));
+        }
+        let _ = match self.get_delta(creds.client, &self.pullUri, &from.commit) {
+            Ok(dir) => Ostree::run(&["static-delta", "apply-offline", &dir])?,
+            Err(_)  => self.pull_commit("agl-remote", creds)?
```

Comment:
Well `REMOTE_PATH` is set to `/etc/ostree/remotes.d/agl-remote.conf`, which is taken from the old ostree scripts. I guess it could be configurable, but this seems fine for now.